### ISSUE TITLE
[BUG] different results with CV when using test_data

### ISSUE THREAD

**Issue by YongSa** [pycaret/pycaret#1445]
**Describe the bug**
When using CV and leaving test_data (setup parameter) undefinied, results are very high (>99% in RF)
When using CV and test_data is being set to a dummy, the results are much worse.

**To Reproduce**

`from sklearn.model_selection import LeaveOneGroupOut` as cv, using groupkfold leads to same results

without test_data: 
`temp_setup = setup(data=temp_train, target="label", silent=True, fold_strategy=cv, fold_groups=temp_train_group, session_id=2021)`

with test_data:
`temp_setup = setup(data=temp_train, target="label", silent=True, fold_strategy=cv, fold_groups=temp_train_group, test_data=temp_dummy, session_id=2021)`

**Expected behavior**
As CV shoulnd't touch test data in model creation, created models should have similar results in accuracy metrics even with and without declaring test_data

**Versions**
2.3.2

**Comment by Yard1** [comment]
if `test_data` is None, the test set will be automatically created by splitting the dataset. Therefore, the amount of rows used for training (including CV) will be smaller.

**Comment by YongSa** [comment]
But even this:
`  clf = setup(data=train, silent=True, target="label", 
              normalize=True,
              fix_imbalance=True,
              fold_strategy=cv,
              fold_groups="group",
              ignore_features=["group"],
              train_size=0.99
              )`

doesn't lead to the same results this one gives me:

`  clf = setup(data=train, silent=True, target="label", 
              normalize=True,
              fix_imbalance=True,
              fold_strategy=cv,
              fold_groups="group",
              ignore_features=["group"],
              test_data=test
              )`

Even thought in both cases the train dataset doesn't contain the test dataset as it has been removed beforehand.
The first setup just uses 1% less data but has results around 95% while the second one has highly volatile results.

**Comment by wkuopt** [comment]
Hi @YongSa,

Is your test_data split in the same way as how it's split by default when test_data is None? Since you are working with imbalanced dataset, I would suggest comparing the distribution of your test_data and train data and see if they are indeed similar.

**Comment by YongSa** [comment]
Hey,

I used a dummy for test_data, so just one line of zeroes with the same columns as the input data. So that would mean I am using 100% of the data in that instance. 

If I set it to none and use a train_split of 0.99, I should be using 99% of the set which shouldn't lead to such different values while validating.

**Comment by wkuopt** [comment]
Thanks @YongSa, could you provide the dataset for us to reproduce your result?

**Comment by YongSa** [comment]
Thank you @wkuopt !

Here is the file:
[wrist.zip](https://github.com/pycaret/pycaret/files/6874467/wrist.zip)

I used Leave One Group Out CV on the column group. Any additional information you need?

**Comment by wkuopt** [comment]
Hi @YongSa, thanks for the file. Could you also provide the full executable code to reproduce the issue? Thanks!

**Comment by YongSa** [comment]
```import pandas as pd
from pycaret.classification import *
from pycaret.utils import enable_colab
from sklearn.model_selection import LeaveOneGroupOut
enable_colab()

# load csv file
train = pd.read_csv(PATH + "wrist.csv", sep=";")
train.reset_index(inplace=True, drop=True)

# create test data as dummy
test = pd.DataFrame(columns=train.columns)
test.loc[0] = 0

# Leave One Group Out CV (LOGO-CV)
cv = LeaveOneGroupOut()

# Train on "train" data with "label" as target, 
# normalize dataset and fix imbalance,
# use LOGO-CV on column "group", 
# ignore column "group" for model creation,
# use "test" as test_data
clf = setup(data=train, silent=True, target="label", 
            normalize=True,
            fix_imbalance=True,
            fold_strategy=cv,
            fold_groups="group",
            ignore_features=["group"],
            test_data=test, # <- best model ridge with F1 of 92.08%, train_size=0.99 <- best model knn with F1 of 100%
            session_id=2021
            )

compare_models(exclude=["ada", "catboost", "lightgbm", "xgboost", "gbc"], 
               sort="F1")
